sources/api/apiclient/src/main.rs

```diff
@@ -776,9 +776,17 @@ async fn run() -> Result<()> {
                 }
             };
 
-            set::set(&args.socket_path, settings)
-                .await
-                .context(error::SetSnafu)?;
+            match set::set(&args.socket_path, settings).await {
+                Ok(result) => result,
+                Err(e) => match e {
+                    set::Error::Raw { source: _ } => {
+                        return Err(error::Error::Raw {
+                            source: Box::new(e),
+                        })
+                    }
+                    _ => return Err(error::Error::Set { source: e }),
+                },
+            };
```

Comment:
extreme nit -- `map_err` can help clarify this sort of code. Not that it took more than a few seconds of brain power, but I was originally trying to figure out why we needed to extract `Ok(result) => result`.

```suggestion
            set::set(&args.socket_path, settings).await
                .map_err(|e| {
                    match e {
                        set::Error::Raw { source: _ } => {
                            error::Error::Raw { source: Box::new(e) }
                        },
                        _ => error::Error::Set { source: e },
                    }
                })?;
            };
```

Rust code generally tries to avoid explicit `return` statements as well, usually hidden behind `?` in error handling code.